src/software/sensor_fusion/sensor_fusion.cpp

```diff
@@ -180,6 +180,55 @@ void SensorFusion::updateWorld(
     }
 }
 
+static double computeDistanceBetweenBallAndRobot(const Robot &robot, const Ball &ball)
+{
+    Vector robot_position = {robot.position().x(), robot.position().y()};
+    Vector ball_position  = {ball.position().x(), ball.position().y()};
+
+    // using the classic distance formula
+    return (ball_position - robot_position).length();
+}
```

Comment:
I'm not sure we need this function -- we've overridden a couple of operators to do this inline easily

see below

Reply:
resolved: I did not know that existed.